### PR TITLE
Adding Support For Arbitrary Executables to Bench Test Scenarios

### PR DESCRIPTION
**Pull request by simpsont-oci**

### CODE REVIEW

File: performance-tests/bench/test_controller/main.cpp [hunk 1]
```diff
@@ -250,6 +250,8 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
     try {
       unsigned long max_value = 0;
       for (const std::string& i : get_dir_contents(results_path)) {
+        if (i.find_first_not_of("0123456789") != std::string::npos)
+          continue;
```
Comment (mitza-oci) [inline performance-tests/bench/test_controller/main.cpp:254]:
```suggestion
        if (i.find_first_not_of("0123456789") != std::string::npos) {
          continue;
        }
```

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: performance-tests/bench/common/util.cpp [hunk 2]
```diff
@@ -47,7 +60,18 @@ unsigned get_option_argument_uint(int& i, int argc, ACE_TCHAR* argv[])
...
+{
+  size_t pos = input.find(oldstr);
+  while (pos != std::string::npos) {
+    input.replace(pos, oldstr.size(), newstr);
+    pos = input.find(oldstr);
+  }
+  return input;
+}
+
+std::string join_path(const std::string& arg)
```
[the remark this answers is not in the document]
Comment (simpsont-oci) [reply performance-tests/bench/common/util.cpp:73]: This is the the n=1 case for the varargs template version of join_path in util.h.

File: performance-tests/bench/common/util.cpp [hunk 3]
```diff
@@ -9,7 +9,20 @@
...
+#ifdef __GNUC__
+#  pragma GCC diagnostic push
+#  if defined(__has_warning)
+#    if __has_warning("-Wdeprecated-declarations")
+#      pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#    endif
+#  elif __GNUC__ > 7
+#    pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#  endif
+#endif
```
[the remark this answers is not in the document]
Comment (simpsont-oci) [reply performance-tests/bench/common/util.cpp:21]: Should I just make a common DirentWrapper.h in dds/DCPS? I don't think it really hurts anything to include ace/Dirent.h here.

File: performance-tests/bench/common/util.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mitza-oci) [reply performance-tests/bench/common/util.cpp:73]: OK, I suggest making it inline so it appears right there in `util.h` and it gets inlined

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**